htlcswitch/switch.go

```diff
@@ -337,54 +345,59 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return err
 		}
 
-		targetLink, err := s.getLinkByShortID(packet.dest)
+		destination, err := s.getLinkByShortID(packet.dest)
 		if err != nil {
 			// If packet was forwarded from another channel link
 			// than we should notify this link that some error
 			// occurred.
-			reason := []byte{byte(lnwire.UnknownDestination)}
+			failure := lnwire.FailUnknownNextPeer{}
+			reason, err := packet.obfuscator.InitialObfuscate(failure)
+			if err != nil {
+				err := errors.Errorf("unable to obfuscate "+
+					"error: %v", err)
+				log.Error(err)
+				return err
+			}
+
 			go source.HandleSwitchPacket(newFailPacket(
 				packet.src,
 				&lnwire.UpdateFailHTLC{
 					Reason: reason,
 				},
-				htlc.PaymentHash, 0,
+				htlc.PaymentHash, 0, true,
 			))
-			err := errors.Errorf("unable to find link with "+
+			err = errors.Errorf("unable to find link with "+
 				"destination %v", packet.dest)
 			log.Error(err)
 			return err
 		}
-		interfaceLinks, _ := s.getLinks(targetLink.Peer().PubKey())
-
-		// Try to find destination channel link with appropriate
```

Comment:
Why was this removed? All links should be checked for their available bandwidth.